Make transformation docstring more precise (#2948)
JuliaData/DataFrames.jl: docs/src/man/split_apply_combine.md

   it must be single name (as a `Symbol` or a string).
6. vectors or matrices containing transformations specified by the `Pair` syntax
   described in points 2 to 5
7. a function which will be called with a `SubDataFrame` corresponding to each group
   when `GroupedDataFrame` is processed or the passed source data frame if

>    `AbstractDataFrame` is processed;

Reviewer: This section only covers split-apply-combine, so the input is supposed to be a `GroupedDataFrame` AFAICT. Though I guess we'd better keep these in sync for consistency?

Author: Yes - this is what I think. Also in general at some point this section should be rewritten at some point not to make such a strong distinction between split-apply-combine and other transformations as currently they are the same conceptually.

>    when `GroupedDataFrame` is processed or the passed source data frame if
>    `AbstractDataFrame` is processed;

Author: ```suggestion
   if a `GroupedDataFrame` is processed, or with the data frame itself if
   an `AbstractDataFrame` is processed;
```
